### PR TITLE
ocb3_done taglen fix

### PR DESCRIPTION
**Pull request by karel-m**
With this patch `ocb3_done(&ocb, tag, &len)` accepts `len` greater that `tag_len` value used in `ocb3_init`.

### CODE REVIEW

**Comment by karel-m** [comment]
You mean a better (strict) `taglen` check in `ocb3_init`?

**Comment by sjaeckel** [comment]
> You mean a better (strict) taglen check in ocb3_init

yep, before this patch it was implicitly limited by the fact that there's no block cipher with more than 128bit blocksize...

File: src/encauth/ocb3/ocb3_init.c [hunk 1]
```diff
@@ -118,6 +118,7 @@ int ocb3_init(ocb3_state *ocb, int cipher,
    if (taglen > (unsigned long)cipher_descriptor[cipher].block_length) {
       taglen = cipher_descriptor[cipher].block_length;
```
Comment (sjaeckel) [inline src/encauth/ocb3/ocb3_init.c]: also I'm not sure anymore if that's correct

Comment (sjaeckel) [inline src/encauth/ocb3/ocb3_init.c]: ah it's even against the spec

> The blockcipher must have a 128-bit blocksize

**Pull request merged**